Code/Tools/ProjectManager/Source/AdjustableHeaderWidget.cpp

```diff
@@ -0,0 +1,118 @@
+        const QVector<QHeaderView::ResizeMode>& resizeModes, QWidget* parent)
+        : QTableWidget(parent)
+    {
+        setObjectName("adjustableHeaderWidget");
+        setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Minimum);
+        setFixedHeight(s_headerWidgetHeight);
+
+        m_header = horizontalHeader();
+        m_header->setDefaultAlignment(Qt::AlignLeft);
+
+        setColumnCount(headerLabels.count());
+        setHorizontalHeaderLabels(headerLabels);
+
+        AZ_Assert(defaultHeaderWidths.count() == columnCount(), "Default header widths does not match number of columns");
+        AZ_Assert(resizeModes.count() == columnCount(), "Resize modesdoes not match number of columns");
+
+        for (int column = 0; column < columnCount(); ++column)
+        {
+            m_header->resizeSection(column, defaultHeaderWidths[column]);
+            m_header->setSectionResizeMode(column, resizeModes[column]);
+        }
+
+        m_header->setMinimumSectionSize(minHeaderWidth);
+        m_header->setCascadingSectionResizes(true);
+
+        connect(m_header, &QHeaderView::sectionResized, this, &AdjustableHeaderWidget::OnSectionResized);
+    }
+
+    void AdjustableHeaderWidget::OnSectionResized(int logicalIndex, int oldSize, int newSize)
+    {
+        const int headerCount = columnCount(); 
+        const int headerWidth = m_header->width();
+        const int totalSectionWidth = m_header->length();
+
+        if (totalSectionWidth > headerWidth && newSize > oldSize)
+        {
+            int xPos = 0;
+            int requiredWidth = 0;
+
+            for (int i = 0; i < headerCount; i++)
+            {
+                if (i < logicalIndex)
+                {
+                    xPos += m_header->sectionSize(i);
+                }
+                else if (i == logicalIndex)
+                {
+                    xPos += newSize;
+                }
+                else if (i > logicalIndex)
+                {
+                    if (m_header->sectionResizeMode(i) == QHeaderView::ResizeMode::Fixed)
+                    {
+                        requiredWidth += m_header->sectionSize(i);
+                    }
+                    else
+                    {
+                        requiredWidth += m_header->minimumSectionSize();
+                    }
+                }
+            }
+
+            if (xPos + requiredWidth > headerWidth)
+            {
+                m_header->resizeSection(logicalIndex, oldSize);
+            }
+        }
+
+        // wait till all columns resized
+        QTimer::singleShot(0, [&]()
+        {
+            // only re-paint when the header and section widths have settled
+            const int headerWidth = m_header->width();
+            const int totalSectionWidth = m_header->length();
+            if (totalSectionWidth == headerWidth)
+            {
+                emit sectionsResized();
+            }
+        });
```

Comment:
Might be able to call `updateGeometry()` and then do this logic immediately instead of having to delay until the next even loop tick

Reply:
I may not fully understand where you recommend calling `updateGeometry()` here, but I believe I tried some method of this without success.  When a user resizes a header, you always get multiples of these `sectionresized` signals - one per header that had a size change, and there isn't a signal for when all resizing is done.  The `singleShot()` is really to ensure we are signaling the columns to repaint/update after all the `sectionresized` signals are done and the header section sizes are stable.  If we don't, the columns do a fun little jiggle animation.

Reply:
Is that because of the cascading resizes flag? Hmm yeah it might work if you only called `updateGeometry` once, but if it does indeed need to be queued then I'm OK with this.

In general, I try to avoid having to delay things until the next event tick because it can lead to tricky/cascading issues later on, but I can appreciate that there are lots of scenarios where they are necessary because of Qt limitations.